chapters/cli/exit.rst

```diff
@@ -1,27 +1,29 @@
 ====================
 Gracefully Exit Jina
 ====================
-In this section, you will learn best practices to shut down a Flow and exit Jina correctly.
+
+In this section, you will learn best practices shutting down a Flow and exiting Jina correctly.
 
 .. contents:: Table of Contents
     :depth: 4
 
 Feature description and expected outcome
 ----------------------------------------
-Jina provides several methods to exit gracefully.
-Through these ways the Flow will terminate and release all resources.
+Jina provides several methods to exit this service gracefully. In this way, the Flow is terminated and all occupied resources are released.
 
 Before you start
 -----------------
-Make sure you install Jina via `Installation <https://docs.jina.ai/chapters/install/os/index.html>`_.
+Make sure that Jina is installed properly as explained in `Installation <https://docs.jina.ai/chapters/install/os/index.html>`_. Define a Flow as explained in `101: Basic components <https://docs.jina.ai/chapters/101/>` , and `102: How basic components work together <https://docs.jina.ai/chapters/102/>`_.
 
 Implementation
 ---------------
 
 In Python
 ^^^^^^^^^
 
-If you use ``with`` scope to start a Flow, then all resources (including Pods of all kinds) of the Flow will be released after you move out from the scope.
+A common way to start a Flow in your Python code is using a ``with`` statement. Moving out from the ``with`` scope, all resources (including Pods of all kinds) of the Flow will be released immediately. 
+
+The source snippet below demonstrates this by defining a Flow named ``f`` containing the two names ``p1`` and ``p2``, only. Next, the ``with`` statement simply contains a ``pass`` statement. The execution of the ``pass`` statement is followed by leaving the scope of the ``with`` statement, and releasing the resources for ``f``.
```

Comment:
two names == two **pods** named p1 and p2. 

I would remove the section about the pass statement, I think that's not relevant to the user understanding of the problem.

Reply:
OK, I will shorten the paragraph.